docs/architecture/adr-050-sign-mode-textual.md

```diff
@@ -51,11 +51,71 @@ This also prevents users signing over any hashed transaction data (fee, transact
 
 We propose to maintain functional tests using bijectivity in the SDK.
 
-### 2. Only ASCII 32-127 characters allowed
+### 2. UTF-8 characters allowed, but signing devices MAY convert them before display
 
-Ledger devices have limited character display capabilities, so all strings MUST only contain ASCII characters in the 32-127 range.
+The SIGN_MODE_TEXTUAL specification allows all UTF-8 characters. The textual strings will contain all characters as-is, with the modifications below:
+- the line feed `\n` character (ASCII: 10) is escaped using quotation marks: `"\n"`. This is to disambiguate with the `\n` control character used to signal a screen change on the signing device.
+- the quotation mark character `"` (ASCII: 34) is escaped with a backslash prefix: `\"`. This is to allow bijectivity if the signing device decides to convert UTF-8 characters into its own set of displayable characters, using `"` as a control character.
+
+For example, consider the transaction's `memo` field:
+
+```
+// JSON: {"memo": "foo"}
+Memo: foo
+
+// JSON: {"memo": "\"foo\""}
+Memo: \"foo\"
```

Comment:
Since we don't use quotation marks as metacharacters, we could also just say the following without risk of ambiguity:

```
Memo: "foo"
```